akka-sample-sharding-scala/src/main/scala/sample/sharding/Protocol.scala

```diff
@@ -0,0 +1,58 @@
+package sample.sharding
+
+import akka.actor.typed.ActorRef
+
+object Protocol {
+
+  trait Command extends CborSerializable
```

Comment:
we should see if we can narrow down the protocol for each actor instead of all using this shared Command. 
For example Guardian should have it's own `Command` and `GetWeatherStations` only implementing that, since a TemperatureDevice shouldn't know about that.

There could be a few that must implement several to facilitate easy delegation without converting, such as UpdateDevice that is handled by both Guardian and then forwarded to TemperatureDevice

Reply:
Agreed, something I still wanted to do.